backend/runner/runner.go

```diff
@@ -568,3 +595,27 @@ func (s *Service) healthCheck(writer http.ResponseWriter, request *http.Request)
 	}
 	writer.WriteHeader(http.StatusServiceUnavailable)
 }
+
+func (s *Service) startPgProxy(ctx context.Context, module *schema.Module, started chan<- pgproxy.Started) error {
+	logger := log.FromContext(ctx)
+
+	databases := map[string]*schema.Database{}
+	for _, decl := range module.Decls {
+		if db, ok := decl.(*schema.Database); ok {
+			databases[db.Name] = db
+		}
+	}
+
+	if err := pgproxy.New(":0", func(ctx context.Context, params map[string]string) (string, error) {
+		db, ok := databases[params["database"]]
+		if !ok {
+			return "", fmt.Errorf("database %s not found", params["database"])
+		}
+		logger.Debugf("Resolved DSN (%s): %s", params["database"], db.Runtime.DSN)
```

Comment:
Here is where the logic to use a dsn resolver based on provisioned runtime (like `type: aws-iam-auth`) will be added